newIDE/app/src/EventsSheet/ParameterFields/ObjectAnimationNameField.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import React, { Component } from 'react';
+import GenericExpressionField from './GenericExpressionField';
+import { type ParameterFieldProps } from './ParameterFieldCommons';
+import { type ExpressionAutocompletion } from '../../ExpressionAutocompletion';
+import getObjectByName from '../../Utils/GetObjectByName';
+import { getLastObjectParameterValue } from './ParameterMetadataTools';
+import { mapFor } from '../../Utils/MapFor';
+
+const gd: libGDevelop = global.gd;
+
+export default class ObjectAnimationNameField extends Component<
+  ParameterFieldProps,
+  void
+> {
+  _field: ?GenericExpressionField;
+
+  focus() {
+    if (this._field) this._field.focus();
+  }
+
+  getAnimationNames(
+    props: ParameterFieldProps
+  ): Array<ExpressionAutocompletion> {
+    const {
+      project,
+      scope,
+      instructionMetadata,
+      instruction,
+      expressionMetadata,
+      expression,
+      parameterIndex,
+    } = props;
+
+    const objectName = getLastObjectParameterValue({
+      instructionMetadata,
+      instruction,
+      expressionMetadata,
+      expression,
+      parameterIndex,
+    });
+    if (!objectName || !project) {
+      return [];
+    }
+
+    const object = getObjectByName(project, scope.layout, objectName);
+    if (!object) {
+      return [];
+    }
+
+    const spriteObject = gd.asSpriteObject(object);
+    if (!spriteObject) {
```

Comment:
Unfortunately `castObject` is a "dumb" cast (some kind of "reinterpret_cast" in C++) that will always give you back an object, even if it does not make sense.
Instead, check the type:

```
if (object.getType() == "Sprite") {
  const spriteObject = gd.asSpriteObject(obejct);
  // Do the rest
}

return [];
```

Exceptionnally, avoid an early return here because we might need to add more object types in the future, or convert this to a "capability" of the object to have animations. Not using an early return is a good sign that this can be extended in the future for other object types or more generally for any object having animations.